getinput/getinput.go

```diff
@@ -31,8 +35,39 @@ func Get(year int, day int, sessionKey string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	return contents, nil
+}
+
+func Get(year int, day int, sessionKey string) ([]string, error) {
+	cacheFilename := filepath.Join(os.TempDir(), "advent", strconv.Itoa(day))
+
+	var b []byte
+	if _, err := os.Stat(cacheFilename); os.IsNotExist(err) {
+		if sessionKey == "" {
+			return nil, errors.New("no session key provided")
+		}
+		err := os.MkdirAll(filepath.Dir(cacheFilename), 0777)
```

Comment:
Use the stored constant for this
`err := os.MkdirAll(filepath.Dir(cacheFilename), os.ModePerm)`